src/Microsoft.DotNet.Wpf/src/Shared/MS/Internal/AppDomainShutdownMonitor.cs

```diff
@@ -30,57 +30,53 @@ static AppDomainShutdownMonitor()
         {
             AppDomain.CurrentDomain.DomainUnload += OnShutdown;
             AppDomain.CurrentDomain.ProcessExit += OnShutdown;
-
-            _dictionary = new Dictionary<WeakReference, WeakReference>();
         }
-        
-        public static void Add(WeakReference listener)
+
+        public static void Add(WeakReference<IAppDomainShutdownListener> listener)
         {
-            Debug.Assert(listener.Target != null);
-            Debug.Assert(listener.Target is IAppDomainShutdownListener);
+            Debug.Assert(listener.TryGetTarget(out _));
```

Comment:
I don't see why this check is necessary or even valid, as the target can become null any time after...

Reply:
@miloush Thank you and it is just a debug code